pkg/game/board.go

```diff
@@ -222,3 +222,138 @@ func (board *board) checkCompleted(
 	}
 	return scoreReport, nil
 }
+
+/*
+It analyzes road directed by roadSide parameter.
+It doesn't analyze starting tile.
+returns: road_finished, score, [meeples on road], loop, sideFinishedOn
+param roadSide: always indicates only one cardinal direction!
+*/
+func (board *board) CheckRoadInDirection(roadSide side.Side, startTile elements.PlacedTile) (bool, int, []elements.Meeple, bool, side.Side) {
+	var meeples = []elements.Meeple{}
+	var tile = startTile
+	var tileExists bool
+	var score = 0
+	var road *elements.PlacedFeature
+	var finished bool
+	// check finished on way
+	// do while loop
+	for {
+		tile, tileExists = board.GetTileAt(tile.Position.Add(elements.PositionFromSide(roadSide)))
+		roadSide = roadSide.ConnectedOpposite()
+		// check if tile exists or loop
+		if !tileExists || tile.Position == startTile.Position {
+			// tile does not exist
+			// finish
+			break
+		}
+
+		score++
+
+		// Get road feature
+		road = tile.GetPlacedFeatureAtSide(roadSide)
+
+		// check if there is meeple on the feature
+		if road.MeepleType != elements.NoneMeeple {
+			meeples = append(meeples, road.Meeple)
+		}
+
+		if road.Sides.GetCardinalDirectionsLength() == 1 {
+			// found the end of a road
+			break
+		}
+		// swap to other end of the road on the same tile
+		if road.Sides.GetNthCardinalDirection(0) == roadSide {
+			roadSide = road.Sides.GetNthCardinalDirection(1)
+		} else {
+			roadSide = road.Sides.GetNthCardinalDirection(0)
+		}
+
+	}
+	finished = tileExists && ((road.Sides.GetCardinalDirectionsLength() == 1) || (tile.Position == startTile.Position))
+
+	looped := (tile.Position == startTile.Position)
+	return finished, score, meeples, looped, roadSide
+}
+
+/*
+Calculates score for road.
+
+returns: ScoreReport, checked sides of the start tile (also including loop)
+*/
+func (board *board) ScoreRoadCompletion(tile elements.PlacedTile, road feature.Feature) (elements.ScoreReport, side.Side) {
+	var meeples = []elements.Meeple{}
+	var leftSide, rightSide side.Side
+	var score = 1
+	leftSide = road.Sides.GetNthCardinalDirection(0)
+	rightSide = road.Sides.GetNthCardinalDirection(1)
```

Comment:
If I understand correctly, these sides will not always be left and right, but for example top and left or something. I'd suggest adding a comment to explain it (or maybe changing the variable names, but I think comment would be enough)

Reply:
It's much easier to ilustrate with those names, it is supposed to indicate both ends, but naming it first and second end may lead to harder to follow the code.